Dosemu tur

stsp opened the pull request:
Hi guys!

This is my attempt to build dosemu2
under termux. I haven't tried dosemu1
for that task, as it doesn't support aarch64,
whereas dosemu2 does.
Unfortunately, dosemu2 is a monster,
(while dosemu1 is very small), so I spent
almost a month figuring out its deps and
packing it up into termux... I don't even
know if people will appreciate that amount
of new packages, all of which are just a
build-deps. :( I actually only added the
most essential deps - it had much more,
I removed some.

Status:
- Builds on CI: from termux, from tur and the hacked-up my own
- Tested on waydroid/x86_64 only, so please test on aarch64!
- All acceleration (jit, kvm, native execution on x86 host) - disabled
- pthread_cancel() replaced with pthread_kill(), so on exit you'll see messages like "terminated on signal" etc

I'd say its a pure miracle that it even works.
But I believe it suits termux better than
dosbox does, because it can flawlessly work
on text terminal without gfx mode.
Which termux is all about. :)

stsp commented:
Here's the demo of dosemu2 on terminal:
https://www.youtube.com/watch?v=_seqxpqvuXs
As can be seen, its not just for simple text -
most of complex TUI things actually work.

stsp commented:
Oh, can someone please tell me what
is this "dry-run" thing? I only made sure
the packages can be built.

stsp commented:
I removed the `.git` suffixes from URLs.
I suspect this is what makes dry-run to
fail... But only the trial can tell.

licy183 commented on tur/comcom64/build.sh:
```diff
@@ -0,0 +1,28 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION="0.3"
+TERMUX_PKG_SRCURL=git+https://github.com/dosemu2/comcom64
+TERMUX_PKG_GIT_BRANCH=master
```
`master` it not valid. If it uses thunk version, you should checkout specific commit in `termux_step_post_get_source`. See the example below.

https://github.com/termux-user-repository/tur/blob/7e0f324ef3a38551182b9e68c134753586064a31/tur/vkmark/build.sh#L15-L36.

stsp replied:
Do I really need to do this, or maybe
just putting commit sha as branch is
enough?

stsp replied:
Also I see packages specify
`TERMUX_PKG_SHA256` in case of
the particular commit. What kind
of sha is this? I only know how to
calc sha of a release zip file.

stsp replied:
Done.

licy183 commented on tur/fdpp/build.sh:
```diff
@@ -0,0 +1,26 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION=1.8
+TERMUX_PKG_SRCURL=git+https://github.com/dosemu2/fdpp
+TERMUX_PKG_GIT_BRANCH=master
```
Ditto

licy183 commented on tur/instfd/build.sh:
```diff
@@ -0,0 +1,14 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION=0.1
+TERMUX_PKG_SRCURL=git+https://github.com/dosemu2/install-freedos
+TERMUX_PKG_GIT_BRANCH=master
```
Ditto

licy183 commented on tur/nasm-segelf/build.sh:
```diff
@@ -0,0 +1,18 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION="2.16.01"
+TERMUX_PKG_SRCURL=git+https://github.com/stsp/nasm-segelf
+TERMUX_PKG_GIT_BRANCH=master
```
Ditto

licy183 commented on tur/dj64dev/build.sh:
```diff
@@ -0,0 +1,25 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION=0.2
+TERMUX_PKG_SRCURL=git+https://github.com/stsp/dj64dev
+TERMUX_PKG_GIT_BRANCH=master
```
Ditto

licy183 commented on tur/djstub/build.sh:
```diff
@@ -0,0 +1,31 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION=0.2
+TERMUX_PKG_SRCURL=git+https://github.com/stsp/djstub
+TERMUX_PKG_GIT_BRANCH=main
```
Ditto

licy183 commented on tur/dosemu2/build.sh:
```diff
@@ -0,0 +1,17 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION="2.0pre9-git"
+TERMUX_PKG_SRCURL=git+https://github.com/dosemu2/dosemu2
+TERMUX_PKG_GIT_BRANCH=devel
```
Ditto

licy183 commented on tur/thunk-gen/build.sh:
```diff
@@ -0,0 +1,49 @@
+TERMUX_PKG_MAINTAINER="@stsp"
+TERMUX_PKG_VERSION=1.2
+TERMUX_PKG_SRCURL=git+https://github.com/stsp/thunk_gen
+TERMUX_PKG_GIT_BRANCH=main
```
Ditto

licy183 commented on tur/ctags-cross/build.sh:
```diff
@@ -0,0 +1,41 @@
+TERMUX_PKG_DESCRIPTION="Universal ctags: Source code index builder"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux-user-repository"
+TERMUX_PKG_VERSION="3-6.1.0"
```
```suggestion
TERMUX_PKG_VERSION="6.1.0"
```

stsp replied:
Since ctags is also present in termux,
I set it to "2:6.1.0" now to match termux.

licy183 commented:
As you're the maintainer of these packages, I'll merge this PR after these things are get resolved. Thanks for your contribution!

stsp commented:
> As you're the maintainer of these packages, I'll merge this PR after these things are get resolved.

I addressed your comments.
As for maintaining - I only tested
these packages under waydroid/x86_64.
If setting the maintenance to
`@termux-user-repository` helps
better testing, I can do that.

Status update:
- enabled jit on x86 platforms
- enabled LTO

So mostly this is a fully-functional build,
just with GUI stripped off (termux doesn't
need GUI, it needs TUI). It seems, for arm
platforms there is no JIT even in upstream
version, so for now this is probably the
best what I could package.

licy183 commented:
LGTM. Thanks!
